### ISSUE TITLE
Query plugin development

### ISSUE THREAD

**Issue by PatriceMoreau56** [opensourceBIM/BIMserver#670]
Hi,

As we can't use the BimQL plugin anymore, I planned to develop my own query plugin.
I tried to make an "Hello world" query plugin but i can't find the correct JSON request to "call" my plugin. It seems that in old version of BIMserver the download method had a downloadType parameter that is not used anymore. Can you help me ?
By the way, according to your [wiki page](https://github.com/opensourceBIM/BIMserver/wiki/Deprecated#query-plugin-interface), can you confirm that it's useless to try to develop a query plugin or a new interface for query plugin development is planned?

Thank you

**Comment by rubendel** [comment]
What specific features are you missing in the current embedded query implementation?

**Comment by PatriceMoreau56** [comment]
The specific feature i need is similar to the issue #675.
I created a custom Pset, let's name it "customPset". This Pset contains several Properties of different types, to have a simple case give to "customPset" 3 properties _P1_, _P2_, _P3_ of type IfcPropertySingleValue.
I rely this Pset on different objects throw an IfcRelDefinesByProperties.
What I want is : 
1) In one case, with all the objects (or a specific type) that have a _P1_ property (whatever the value) retrieve the globalId of the object, and the values of the properties _P1_, _P2_ and _P3_
2) In another case, with all the objects (or a specific type) that have a "customPset", retrieve the globalId of the object, and the values and the name of all the properties of the "customPset"

That's why a tried to do this with a query plugin, to avoid multiply the network requests and manipulate the model directly on the server.
I saw you open the #678 issue which is closed to my needs.
